src/data/markdown/translated-guides/en/05 Test Types/03 Stress testing.md

```diff
@@ -46,153 +46,179 @@ We recommend running a stress test in a UAT or staging environment.
+Here's an example script configuration for a spike test.
+As with stress testing, `ramping-arrival-rate` is a good executor for a spike test.
 
 <CodeGroup labels={["Spike test k6 example"]} lineNumbers={[true]} heightTogglers={[true]}>
 
 ```javascript
-import http from 'k6/http';
-import { sleep } from 'k6';
+import http from "k6/http";
+import { sleep } from "k6";
 
 export const options = {
-  stages: [
-    { duration: '10s', target: 100 }, // below normal load
-    { duration: '1m', target: 100 },
-    { duration: '10s', target: 1400 }, // spike to 1400 users
-    { duration: '3m', target: 1400 }, // stay at 1400 for 3 minutes
-    { duration: '10s', target: 100 }, // scale down. Recovery stage.
-    { duration: '3m', target: 100 },
-    { duration: '10s', target: 0 },
-  ],
+  scenarios: {
+    spike: {
+      executor: "ramping-arrival-rate",
+      preAllocatedVUs: 1000,
+      timeUnit: "1s",
+      stages: [
+        { duration: "10s", target: 10 }, // below normal load
+        { duration: "1m", target: 10 },
+        { duration: "10s", target: 140 }, // spike to 140 iterations
+        { duration: "3m", target: 140 }, // stay at 140 for 3 minutes
+        { duration: "10s", target: 10 }, // scale down. Recovery stage.
+        { duration: "3m", target: 10 },
+        { duration: "10s", target: 0 },
+      ],
+      gracefulStop: "2m",
+    },
+  },
 };
+
 export default function () {
-  const BASE_URL = 'https://test-api.k6.io'; // make sure this is not production
+  const BASE_URL = "https://test-api.k6.io"; // make sure this is not production
 
   const responses = http.batch([
-    ['GET', `${BASE_URL}/public/crocodiles/1/`, null, { tags: { name: 'PublicCrocs' } }],
-    ['GET', `${BASE_URL}/public/crocodiles/2/`, null, { tags: { name: 'PublicCrocs' } }],
-    ['GET', `${BASE_URL}/public/crocodiles/3/`, null, { tags: { name: 'PublicCrocs' } }],
-    ['GET', `${BASE_URL}/public/crocodiles/4/`, null, { tags: { name: 'PublicCrocs' } }],
+    [
+      "GET",
+      `${BASE_URL}/public/crocodiles/1/`,
+      null,
+      { tags: { name: "PublicCrocs" } },
+    ],
+    [
+      "GET",
+      `${BASE_URL}/public/crocodiles/2/`,
+      null,
+      { tags: { name: "PublicCrocs" } },
+    ],
+    [
+      "GET",
+      `${BASE_URL}/public/crocodiles/3/`,
+      null,
+      { tags: { name: "PublicCrocs" } },
+    ],
+    [
+      "GET",
+      `${BASE_URL}/public/crocodiles/4/`,
+      null,
+      { tags: { name: "PublicCrocs" } },
+    ],
   ]);
-
-  sleep(1);
 }
 ```
 
 </CodeGroup>
 
 The VU chart of a spike test should look similar to this:
-![Virtual user chart of a spike test](./images/spike-test.png)
-
-Note, the test starts with a period of 1 minute of low load, a quick spike to very high load, followed by a recovery period of low load.
+![Virtual user chart of a spike test](./images/spike-test.png "The test starts with a period of 1 minute of low load, a quick spike to very high load, followed by a recovery period of low load.")
```

Comment:
Unfortunately k6 at this point (IIRC) reports all the allocated VUs as constnatly in use even if thre are doesn't require it. So if you have allocated 1000 VUs and need 1 to do  1 IPS it will still report 1000. Mostly because it in practice reports how many are active more then anything else. So this graph is ... misleading and I am not certain is needed.